src/ipyradiant/visualization/vis_tool.py

```diff
@@ -0,0 +1,114 @@
+import os
+import textwrap
+from pathlib import Path
+from rdflib import URIRef, Literal, Graph, BNode
+import traitlets as T
+from rdflib.namespace import RDF
+from pandas import DataFrame
+import ipywidgets as W
+from ipycytoscape import MutableDict, CytoscapeWidget
+
+default_edge = {"selector": "edge", "css": {"line-color": "blue"}}
+
+default_node = {"selector": "node", "css": {"background-color": "grey"}}
+
+
+class RDFVisualization(W.GridBox):
+    graph = T.Instance(Graph, allow_none=True)
+    cyto_widget = T.Instance(CytoscapeWidget, allow_none=True)
+    nodes = T.List()
+    click_output = T.Instance(W.Output)
+    click_output_box = T.Instance(W.VBox)
+    selected_node = T.Unicode(allow_none=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.output_title = W.HTML("<h3>Output from Clicks</h3>")
+        self.layout = W.Layout(grid_template_columns="60% 40%", height="500px")
+        self.click_output_box.children = [self.output_title, self.click_output]
+        self.children = [self.cyto_widget, self.click_output_box]
+
+    @T.default("click_output")
+    def _make_click_output(self):
+        return W.Output()
+
+    @T.default("nodes")
+    def _make_default_nodes(self):
+        return []
```

Comment:
I think you can just pass `default_value=[]` when you initialize the nodes trait on line 19

Reply:
non-scalars are funny... ths may be appropriate in this case. you don't want a global list.